geode-core/src/test/java/org/apache/geode/management/internal/FederatingManagerTest.java

```diff
@@ -437,14 +474,79 @@ public void removeMemberArtifactsDoesNotRemoveAllProxiesIfMonitoringRegionIsNull
         .thenReturn(mock(Region.class));
     when(system.getDistributedMember())
         .thenReturn(member);
-    FederatingManager federatingManager = new FederatingManager(repo, system, service, cache,
-        statisticsFactory, statisticsClock, proxyFactory, messenger, executorService);
+    FederatingManager federatingManager =
+        new FederatingManager(repo, system, service, cache, statisticsFactory,
+            statisticsClock, proxyFactory, messenger, cancelCriterion, executorService);
 
     federatingManager.removeMemberArtifacts(member, false);
 
     verifyNoMoreInteractions(proxyFactory);
   }
 
+  @Test
+  public void removeMemberWaitsForStartManager() throws Exception {
+    CountDownLatch latch = new CountDownLatch(1);
+    CyclicBarrier barrier = new CyclicBarrier(2);
+    ExecutorService executorService = mock(ExecutorService.class);
+    List<Future<InternalDistributedMember>> futureTaskList = Collections.emptyList();
+
+    when(executorService.invokeAll(any())).thenAnswer(invocation -> {
+      awaitCyclicBarrier(barrier);
+      awaitCountDownLatch(latch);
+      return futureTaskList;
+    });
+
+    FederatingManager federatingManager =
+        spy(new FederatingManager(repo, system, service, cache, statisticsFactory, statisticsClock,
```

Comment:
Does this need to be a spy? I don't see any uses of the spyness (spyity?) in this test.

Reply:
Removing use of spy. The test no longer does any spying.

Reply:
Spy is gone!